wagtail/contrib/frontend_cache/backends/cloudfront.py

```diff
@@ -45,7 +45,8 @@ def __init__(self, params):
                 self.hostnames = list(self.cloudfront_distribution_id.keys())
 
     def purge_batch(self, urls):
-        paths_by_distribution_id = defaultdict(set)
+        # Changed: use a list to preserve insertion order
+        paths_by_distribution_id = defaultdict(list)
```

Comment:
This is a potential performance regression - each run of `if path not in paths_by_distribution_id[distribution_id]` will have to sequentially scan the list, so this will end up performing n^2 operations in the number of items, which could be significant in a long list.

(Also, "Changed:" in a comment is unnecessary. Someone reading the code is only concerned with the current state - if they want to know about the change history, they can always look that up in Github.)